### ISSUE TITLE
[BUG] Text-generation strategy for string arguments includes surrogate characters

### ISSUE THREAD

**Issue by dkbarn** [Stranger6667/hypothesis-graphql#30]
I'm trying to figure out why I keep getting the following error: ```UnicodeEncodeError: 'utf-8' codec can't encode characters in position 0-1: surrogates not allowed```

For some reason, the text strategy for arguments of type String is generating unicode data that cannot be encoded as UTF-8, because it includes surrogate characters.

From what I can tell, hypothesis-graphql is using ```hypothesis.strategies.text()``` for String-based arguments: https://github.com/Stranger6667/hypothesis-graphql/blob/master/src/hypothesis_graphql/_strategies/primitives.py#L46

The documentation for ```hypothesis.strategies.text()``` explicitly states:
> The default alphabet strategy can generate the full unicode range but excludes surrogate characters because they are invalid in the UTF-8 encoding.

https://hypothesis.readthedocs.io/en/latest/data.html#hypothesis.strategies.text

So I'm very confused why surrogate characters are being produced.

**To Reproduce**
Run this test using pytest. You should get a UnicodeEncodeError. You can include ```--log-cli-level=INFO``` to see all the values being generated for the ```user``` argument.

```
import logging
import graphql
import schemathesis

schema = schemathesis.graphql.from_file(
    """
    type Query {
        hello(user: String!): String
    }
    """
)

logger = logging.getLogger()

@schema.parametrize()
def test_schema(case):
    document = graphql.parse(case.body)
    argument_node = document.definitions[0].selection_set.selections[0].arguments[0]
    if argument_node.name.value == "user":
        value = argument_node.value.value
        logger.info(f"User argument = {value!r}")
        encoded = value.encode("utf8")
```
**Expected behavior**
The text strategy used for String-based arguments includes only characters that can be UTF-8 encoded.

**Environment (please complete the following information):**

- OS: Linux
- Python version: 3.8.5
- schemathesis version: 3.6.1
- hypothesis-graphql version: 0.4.0
- hypothesis version: 6.8.9
- graphql-core version: 3.1.4
- pytest version: 6.2.3

**Comment by dkbarn** [comment]
After some more investigation, I think that ```hypothesis.strategies.text()``` is doing the right thing.

I think that somewhere in the process of doing ```graphql.print_ast()``` -> ```graphql.parse()``` inside ```hypothesis_graphql._strategies.queries.query()``` the unicode values are getting modified to contain surrogate characters. But I'm not sure why.

**Comment by dkbarn** [comment]
I've filed a bug report in graphql-core, because I think this might be an issue with the parser: https://github.com/graphql-python/graphql-core/issues/128

We'll see what they say.

**Comment by Stranger6667** [comment]
Interesting behavior. Thank you for opening this :) Indeed, let's wait for what the `graphql-core` team will say about it.

**Comment by dkbarn** [comment]
I'm discussing with the dev on graphql-core about the possibility of adding support for the full unicode range. He wants to wait for graphql-js to do this first, and then he'll port it to the python library. So it's probably not something that will happen soon.

In the meantime, the official GraphQL spec does not currently support any unicode character above U+FFFF, and the graphql-core library has undefined behaviour when such a character occurs in a GraphQL document. Given this, it seems like the logical choice for hypothesis-graphql is to not generate characters that the spec doesn't support. This would mean passing ```max_codepoint=0xffff``` to the text strategy that it uses. What do you think?

**Comment by Stranger6667** [comment]
Great! Makes total sense to me

**Comment by Stranger6667** [comment]
@dkbarn FYI, the fix is released in `0.4.1`. I'll pin it in schemathesis's next version as well
